cluster-controller/src/test/java/io/strimzi/controller/cluster/operations/cluster/KafkaClusterOperationsTest.java

```diff
@@ -111,8 +162,90 @@ private void createCluster(TestContext context, boolean isOpenShift) {
+        ArgumentCaptor<StatefulSet> ssCaptor = ArgumentCaptor.forClass(StatefulSet.class);
+        when(mockSsOps.create(ssCaptor.capture())).thenReturn(Future.succeededFuture());
+
+        ArgumentCaptor<String> metricsCaptor = ArgumentCaptor.forClass(String.class);
+        when(mockCmOps.delete(eq(clusterCmNamespace), metricsCaptor.capture())).thenReturn(Future.succeededFuture());
+
+        when(mockServiceOps.delete(eq(clusterCmNamespace), serviceCaptor.capture())).thenReturn(Future.succeededFuture());
+        when(mockSsOps.delete(anyString(), anyString())).thenReturn(Future.succeededFuture());
+
+        KafkaClusterOperations ops = new KafkaClusterOperations(vertx, openShift,
+                mockCmOps,
+                mockServiceOps, mockSsOps,
+                mockPvcOps);
+
+        // Now try to create a KafkaCluster based on this CM
+        Async async = context.async();
+        ops.delete(clusterCmNamespace, clusterCmName, createResult -> {
+            context.assertTrue(createResult.succeeded());
+
+            Set<String> metricsNames = new HashSet<>();
+            if (kafkaCluster.isMetricsEnabled()) {
+                metricsNames.add(KafkaCluster.metricConfigsName(clusterCmName));
+            }
+            if (zookeeperCluster.isMetricsEnabled()) {
+                metricsNames.add(ZookeeperCluster.zookeeperMetricsName(clusterCmName));
+            }
+            context.assertEquals(metricsNames, new HashSet(metricsCaptor.getAllValues()));
+            verify(mockSsOps).delete(eq(clusterCmNamespace), eq(ZookeeperCluster.zookeeperClusterName(clusterCmName)));
+            //verify(mockServiceOps, times(2)).delete(any(), any());
+
+            context.assertEquals(set(
+                    ZookeeperCluster.zookeeperHeadlessName(clusterCmName),
+                    ZookeeperCluster.zookeeperClusterName(clusterCmName),
+                    KafkaCluster.kafkaClusterName(clusterCmName),
+                    KafkaCluster.headlessName(clusterCmName)),
+                    new HashSet(serviceCaptor.getAllValues()));
+
+            // PvcOperations only used for deletion
+            verifyNoMoreInteractions(mockPvcOps);
+            async.complete();
+        });
+    }
+
+    private ConfigMap getConfigMap(String clusterCmName) {
+        String clusterCmNamespace = "test";
+        int replicas = 1;
+        String image = "bar";
+        int healthDelay = 120;
+        int healthTimeout = 30;
+        String metricsCmJson = metrics ?  "{\"foo\":\"bar\"}" : null;
+        return ResourceUtils.createKafkaClusterConfigMap(clusterCmNamespace, clusterCmName, replicas, image, healthDelay, healthTimeout, metricsCmJson, storage);
+    }
+
     private static <T> Set<T> set(T... elements) {
         return new HashSet<>(asList(elements));
     }
 
+
+    @Test
+    public void testDeleteCluster(TestContext context) {
+        ConfigMap clusterCm = getConfigMap("baz");
+        createCluster(context, clusterCm);
```

Comment:
Rather a question than something you have to change ... but why do you create the cluster first in your test? Since everything is mocked, it should have no impact, or?

Reply:
You're right!